atomics/include/desul/atomics/Generic.hpp

```diff
@@ -140,6 +140,18 @@ struct RShiftOper {
   }
 };
 
+template <class Scalar1, class Scalar2>
+struct BoundedIncOper {
+  DESUL_FORCEINLINE_FUNCTION
+  static Scalar1 apply(const Scalar1& val1, const Scalar2& val2) { return ((val1 >= val2) ? Scalar1(0) : val1 + Scalar1(1)); }
+};
+
+template <class Scalar1, class Scalar2>
+struct BoundedDecOper {
+  DESUL_FORCEINLINE_FUNCTION
+  static Scalar1 apply(const Scalar1& val1, const Scalar2& val2) { return (((val1 == Scalar1(0)) | (val1 > val2)) ? val2 : (val1 - Scalar1(1))); }
```

Comment:
While it *shouldn't* happen, it might be good to have this tweaked to handle negative values.  Probably just make the first comparison `val1 <= ...`.

Reply:
Sounds like a solid idea! This is how RAJA had their build-in atomics behave, so I was just going off of that.

Reply:
Makes sense, in the vast majority of cases it wouldn't matter but I like being defensive about it when we can.

Reply:
I think that this is supposed to model the cuda version found here: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#atomicdec

Seems that it should have the "correct" behavior either way, but it's probably worth asking around to see if changing that ```==``` to a ```<=``` is somehow a horrible idea.